crates/oxc_transformer/src/regexp/mod.rs

```diff
@@ -0,0 +1,246 @@
+
+        let callee = {
+            let symbol_id = ctx.scopes().find_binding(ctx.current_scope_id(), "RegExp");
+            let ident = ctx.create_reference_id(
+                regexp.span,
+                Atom::from("RegExp"),
+                symbol_id,
+                ReferenceFlags::read(),
+            );
+            ctx.ast.expression_from_identifier_reference(ident)
+        };
+
+        let mut arguments = ctx.ast.vec_with_capacity(2);
+        arguments.push(
+            ctx.ast.argument_expression(
+                ctx.ast.expression_string_literal(regexp.span, pattern_source),
+            ),
+        );
+
+        let flags = regexp.regex.flags.to_string();
+        let flags =
+            ctx.ast.argument_expression(ctx.ast.expression_string_literal(regexp.span, flags));
+        arguments.push(flags);
+
+        *expr = ctx.ast.expression_new(
+            regexp.span,
+            callee,
+            arguments,
+            None::<TSTypeParameterInstantiation>,
+        );
+    }
+}
+
+impl<'a> RegExp<'a> {
+    /// Check if the regular expression contains any unsupported flags.
+    fn has_unsupported_regular_expression_flags(&self, flags: RegExpFlags) -> bool {
+        flags.iter().any(|f| match f {
+            RegExpFlags::G | RegExpFlags::I | RegExpFlags::M => true,
+            RegExpFlags::S if self.options.dot_all_flag => true,
+            RegExpFlags::Y if self.options.sticky_flag => true,
+            RegExpFlags::U if self.options.unicode_flag => true,
+            RegExpFlags::D if self.options.match_indices => true,
+            RegExpFlags::V if self.options.set_notation => true,
+            _ => false,
+        })
+    }
+
+    /// Check if the regular expression contains any unsupported syntax.
+    ///
+    /// Based on parsed regular expression pattern.
+    fn has_unsupported_regular_expression_pattern(&self, pattern: &Pattern<'a>) -> bool {
+        // Early return if no unsupported features-related plugins are enabled
+        if !(self.options.named_capture_groups
+            || self.options.unicode_property_escapes
+            || self.options.look_behind_assertions)
+        {
+            return false;
+        }
+
+        pattern.body.body.iter().any(|alternative| {
+            alternative.body.iter().any(|element| match element {
+                Term::CapturingGroup(_) if self.options.named_capture_groups => true,
+                Term::UnicodePropertyEscape(_) if self.options.unicode_property_escapes => true,
+                Term::LookAroundAssertion(assertion)
+                    if self.options.look_behind_assertions
+                        && matches!(
+                            assertion.kind,
+                            LookAroundAssertionKind::Lookbehind
+                                | LookAroundAssertionKind::NegativeLookbehind
+                        ) =>
+                {
+                    true
+                }
+                _ => false,
+            })
```

Comment:
Sorry for barrage of comments!

`Term::Quantifier` contains more nested `Term`s. So it could have one of the `Term`s we're looking for. So that needs some form of recursion/traversal.

`Term::CharacterClass` contains `CharacterClass` which contains `CharacterClassContents` which can contain `UnicodePropertyEscape`, so we need to check for that too.

Reply:
> Sorry for barrage of comments!
> 
> `Term::Quantifier` contains more nested `Term`s. So it could have one of the `Term`s we're looking for. So that needs some kind of recursive checking.
> 
> `Term::CharacterClass` contains `CharacterClass` which contains `CharacterClassContents` which can contain `UnicodePropertyEscape`, so we need to check for that too if `self.options.unicode_property_escapes == true`.

I’ve missed these. I need to look through regexp ast

Reply:
I *think* those are only cases we're missing. I "followed the trail" to the types we're looking for through the RegExp AST, and those are the only other paths I could find to reach them.

Reply:
Now we check against the RegExp AST. I think our test should cover it